### PR TITLE
[circle-mpqsolver] Improve MAE

### PR DESCRIPTION
**Pull request by stamalakhov**
This commit improves accuracy of MAE metric and computes it as mean over mean output errors.

ONE-DCO-1.0-Signed-off-by: s.malakhov <s.malakhov@partner.samsung.com>

### CODE REVIEW

**Review by mhs4670go — APPROVED** [review]
LGTM

File: compiler/circle-mpqsolver/src/core/ErrorMetric.cpp [hunk 1]
```diff
@@ -44,22 +44,28 @@ float MAEMetric::compute(const WholeOutput &first, const WholeOutput &second) co
...
   }
 
-  if (output_size == 0)
+  if (num_output_errors == 0)
   {
     throw std::runtime_error("nothing to compare");
   }
 
-  return error / output_size;
+  return static_cast<double>(output_errors / num_output_errors);
```
Comment (jinevening) [inline compiler/circle-mpqsolver/src/core/ErrorMetric.cpp:70]: Please update function signature too.

Comment (jinevening) [reply compiler/circle-mpqsolver/src/core/ErrorMetric.cpp:70]: If you intended to return `float`, then this line may need to be udpated (`static_cast<float>`).

Comment (stamalakhov) [reply compiler/circle-mpqsolver/src/core/ErrorMetric.cpp:70]: Ahhh. Sorry. Typo!

Comment (stamalakhov) [reply compiler/circle-mpqsolver/src/core/ErrorMetric.cpp]: fixed

**Review by mhs4670go — APPROVED** [review]

**Review by jinevening — APPROVED** [review]

**Pull request merged**